db/sql/sql.go

```diff
@@ -1097,10 +1283,25 @@ func (tx *Transaction) FetchContext(ctx context.Context, s *schema.Schema, filte
 	return list[0], nil
 }
 
+func (tx *CachedTransaction) FetchContext(ctx context.Context, s *schema.Schema, filter transaction.Filter, options *transaction.ViewOptions) (*schema.Resource, error) {
```

Comment:
code duplication with `(tx *Transaction) FetchContext`. don't be shy to refactor the existing code if needed - we've got extensive unit tests to assist you :)